idutils/ext.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of IDUtils
+# Copyright (C) 2024 CERN.
+#
+# IDUtils is free software; you can redistribute it and/or modify
+# it under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Invenio IdUtils module for managing persistent identifiers used in scholarly communication."""
+
+from . import config, detectors, normalizers, utils
+
+
+class IdUtils(object):
+    """Invenio extension."""
+
+    def __init__(self, app=None):
+        """Extension initialization."""
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        """Flask application initialiation."""
+        self.init_config(app)
+        app.extensions["idutils"] = self
+
+        self.init_normalizers(app)
+        self.init_detectors(app)
+        self.init_utils(app)
+
+    def init_config(self, app):
+        """Initialize configuration."""
+        for k in dir(config):
+            if k.isupper():
+                app.config.setdefault(k, getattr(config, k))
+
+    def init_normalizers(self, app):
+        """Initialize normalizers."""
+        app.config["NORMALIZERS"] = {}
+        for k in dir(normalizers):
+            if k.startswith("normalize_"):
+                app.config["NORMALIZERS"].setdefault(k, getattr(normalizers, k))
+
+    def init_detectors(self, app):
+        """Initialize detectors."""
+        app.config["DETECTORS"] = {}
+        for k in dir(detectors):
+            if k.startswith("detect_"):
+                app.config["DETECTORS"].setdefault(k, getattr(detectors, k))
+
+    def init_utils(self, app):
+        """Initialize utility functions."""
+        app.config["IDUTILS"] = {}
+        for k in dir(utils):
+            if k.startswith("is_"):
+                app.config["IDUTILS"].setdefault(k, getattr(utils, k))
```

Comment:
I would not make configurable all the existing, given that they are worldwide recognized "standards". Let's only introduce a new extension point to add your own.